src/Datasource/RulesChecker.php

```diff
@@ -404,4 +413,19 @@ protected function _addError(callable $rule, array|string|null $name = null, arr
 
         return $rule;
     }
+
+    /**
+     * Checks that a rule with the same name doesn't already exist
+     *
+     * @param array<\Cake\Datasource\RuleInvoker> $rules The rules array to check
+     * @param string $name The name to check
+     * @return void
+     * @throws \Cake\Datasource\Exception\RuleAlreadyExistsException
+     */
+    protected function _checkName(array $rules, string $name): void
```

Comment:
```suggestion
    protected function checkName(string $name, array $rules): void
```

We no longer add underscore prefix to protected properties/methods. Also having name as the 1st param feels more natural, that's the order for array_key_exists() too.